packages/truffle-core/lib/console.js

```diff
@@ -52,13 +52,16 @@ class Console extends EventEmitter {
 
     try {
       const abstractions = this.provision();
-      this.repl.start({
-        prompt: "truffle(" + this.options.network + ")> ",
-        context: {
-          web3: this.web3
-        },
-        interpreter: this.interpret.bind(this),
-        done: callback
+      this.web3.eth.getAccounts().then(accounts => {
```

Comment:
Oh shoot. I might be being paranoid, but I think maybe we want this entire function to get wrapped in the `.then()`. I'm not sure, but we might need to make sure that `this.resetContractsInConsoleContext` doesn't rely on some implicit ordering. _shrug_ I hate these callback-style functions everywhere, it makes the code so ugly :(

Reply:
Maybe including this.resetContractsInConsoleContext into .then() will solve the issue?
I don't like this type of callback functions everywhere also :(